src/header/common/host.rs

```diff
@@ -47,43 +49,8 @@ impl Header for Host {
     }
 
     fn parse_header(raw: &[Vec<u8>]) -> ::Result<Host> {
-        from_one_raw_str(raw).and_then(|mut s: String| {
-            // FIXME: use rust-url to parse this
-            // https://github.com/servo/rust-url/issues/42
-            let idx = {
-                let slice = &s[..];
-                let mut chars = slice.chars();
-                chars.next();
-                if chars.next().unwrap() == '[' {
-                    match slice.rfind(']') {
-                        Some(idx) => {
-                            if slice.len() > idx + 2 {
-                                Some(idx + 1)
-                            } else {
-                                None
-                            }
-                        }
-                        None => return Err(::Error::Header) // this is a bad ipv6 address...
-                    }
-                } else {
-                    slice.rfind(':')
-                }
-            };
-
-            let port = match idx {
-                Some(idx) => s[idx + 1..].parse().ok(),
-                None => None
-            };
-
-            match idx {
-                Some(idx) => s.truncate(idx),
-                None => ()
-            }
-
-            Ok(Host {
-                hostname: s,
-                port: port
-            })
+        from_one_raw_str(raw).and_then(|s: String| {
```

Comment:
I believe this step can be skipped (this step to a `String`). I'd expect this entire function could just be `from_one_raw_str(raw)`, since inferrence should tell it to use `<Host as FromStr>`.